### ISSUE TITLE
SSLHandshakeError behind reverse proxy with letsencrypt wildcard certificate

### ISSUE THREAD

**Issue by steckums** [patzly/grocy-android#491]
I have a setup like this to access Grocy:

Containerized Grocy from linuxserver.io's image on a VM exposing port 9001 of the host VM and forwarding it to port 80 of the container.

A completely separate VM from the docker host that runs nginx as a reverse proxy, with the following configuration for the reverse proxy:

```server  {
  listen  443 ssl;
  server_name  grocy.mydomain.com;
  ssl  on;
   proxy_http_version 1.1;
   proxy_set_header X-Real-IP $remote_addr;
   proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
   proxy_set_header X-Forwarded-Proto https;
   location / {
        proxy_pass http://docker-host:9001;
        proxy_set_header Host $http_host;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        }
}
```
And a separate SSL configuration that is applied to all enabled-sites:

```ssl_certificate  /etc/letsencrypt/live/mydomain.com/cert.pem;   
ssl_certificate_key  /etc/letsencrypt/live/mydomain.com/privkey.pem;  
ssl_session_timeout  5m;
ssl_prefer_server_ciphers  on;
ssl_protocols  TLSv1.2 TLSv1.3;
ssl_ciphers  AES256+EECDH:AES256+EDH:!aNULL;
add_header X-Xss-Protection "1; mode=block" always;
add_header X-Content-Type-Options "nosniff" always;
add_header Strict-Transport-Security "max-age=2592000; includeSubdomains" always;
add_header X-Frame-Options "SAMEORIGIN" always;
proxy_hide_header X-Powered-By;
add_header 'Referrer-Policy' 'no-referrer';
```

The reverse proxy VM is responsible for handling all port 80 (upgrades https automatically) and 443 on the network, as configured with pfsense. This VM is also responsible for upgrading the letsencrypt wildcard cert that covers `*.mydomain.com`

Configuring the android app to point to `http://docker-host:9001` on the local network works, as expected, but when trying to connect via `https://grocy.mydomain.com` I receive this error:

``` This server requires a certificate that is not known or trusted by Android.
The server certificate must either be signed by a trusted certificate authority (e.g. letsencrypt.org) or stored in the Android user certificate store of this device.
```

I've not had this issue connecting to any other service through my reverse proxy and my certificate is not expiring soon. By using Chrome to visit `grocy.mydomain.com`, I can see that my site uses TLS1.3, encrypted and authenticated with AES_256_GCM, X25519 as the key exchange, and the cert was issued by R3.

Is there something I am missing here?

**Comment by MrJohnZoidberg** [comment]
Mmh, thanks for your detailed report – Unfortunately, I do not know what this could be caused by... We use Volley as network library (which throws the SSLHandshakeException) and Conscrypt for certificate support but I cannot find anything on the Internet related to this now. I will let this issue open for later, or maybe some other dev has more experience to solve it. :)

**Comment by malteneu** [comment]
Lets encrypt R3 certificates should be signed by ISRG Root X1. If your certificate is signed by DST Root CA X3 it will not work since support expired in September 2021. If this is the case it might just be easy to update the certificate.

Otherwise have you checked wether its possible to access the web version of grocy on your android client? If the problem remains you should try and update your trusted CA. In your settings (commonly under Legal Information) within `Google Play system update licencess` there should be a `Conscrypt` tab. You can check there. Its a bit complicated however.

Otherwiese you can add your WIldcard Certificate within the User Certificate Store on Android

**Comment by steckums** [comment]
I am also not sure how nginx handles chained certificates, but it looks like it is working. When visiting on a PC, I can check the certification path. It does show `ISRG Root X1 > R3 > *.mydomain.com`.

Viewing `grocy.mydomain.com` on my Android client does also show ISRG Root X1 in the Certificate Viewer in the dropdown, and there are no issues accessing the site that way.

**Comment by synonym24** [comment]
Run into the same problem with the images from Docker Hub.
I used the instructions [here](https://github.com/grocy/grocy-docker).

I switched from only the certificate file to the full chain file. After that I could establish a connection without any further issues.

So, I switched from ECC-cert.pem to ECC-fullchain.pem which I'm using now as grocy-nginx.crt.
As I'm using this on a Synology box it's very likely my file names differ from yours.
